### ISSUE TITLE
ansible-lint >= 6.16.0: jinja[invalid] triggered for generator expressions

### ISSUE THREAD

**Issue by nE0sIghT** [ansible/ansible-lint#3633]
##### Summary

With ansible lint starting with 6.16.0 I getting `jinja[invalid]` for generator expressions.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.17.2 using ansible-core:2.14.8 ansible-compat:4.1.5 ruamel-yaml:0.17.32 ruamel-yaml-clib:0.2.7
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Try to lint following playbook:
```
---
- hosts: localhost
  gather_facts: false
  tasks:
    - name: Lint test
      set_fact:
        _somefact: >-
          {{
            {} |
            combine({
              'some': [a for a in someary],
              'other': 'all',
            })
          }}
      vars:
        someary:
          - 1
          - 2
```

##### Desired Behavior

No linting errors should be reported

##### Actual Behavior

```paste below
WARNING  Could not parse missing filter name from error message: template error while templating string: expected token ',', got 'for'. String: {{
  {} |
  combine({
    'some': ,
    'other': 'all',
  })
}}. expected token ',', got 'for'
WARNING  Could not parse missing filter name from error message: template error while templating string: expected token ',', got 'for'. String: {{
  {} |
  combine({
    'some': ,
    'other': 'all',
  })
}}. expected token ',', got 'for'
WARNING  Listing 1 violation(s) that are fatal
jinja[invalid]: template error while templating string: expected token ',', got 'for'. String: {{
  {} |
  combine({
    'some': [a for a in someary],
    'other': 'all',
  })
}}. expected token ',', got 'for'
playbook/tmp.yml:7 Task/Handler: Lint test

Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary               
 count tag            profile rule associated tags 
     1 jinja[invalid] basic   formatting   
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
This is not really a bug, try to run this playbook yourself and you will get a runtime error from ansible. It is true that running `ansible-playbook --syntax-check` will not find this problem.

Unless you find a playbook that ansible accepts to run and linter complains falsely about it, we cannot consider it a valid bug.

Hint: jinja looks like python, but is not python!
